scalafix/rules/src/main/scala/fix/FixBqSaveAsTable.scala

```diff
@@ -11,28 +11,50 @@ class FixBqSaveAsTable extends SemanticRule("FixBqSaveAsTable") {
 
   override def fix(implicit doc: SemanticDocument): Patch = {
     doc.tree.collect {
-      case a @ Term.Apply(fun, head :: tail) =>
+      case a @ Term.Apply(fun, params) =>
         if (fun.symbol.normalized.toString.contains(methodName)) {
           fun match {
             case Term.Select(qual, name) =>
               name match {
                 case Term.Name("saveAvroAsBigQuery") if expectedType(qual, scoll) =>
-                  // the rest of args should be named because the parameter order has changed
-                  if (
-                    tail.exists(!_.toString.contains("=")) ||
-                    // `avroSchema` doesn't exist in the list of the new method
-                    tail.exists(_.toString.contains("avroSchema"))
-                  ) {
-                    Patch.empty // not possible to fix, leave it to `LintBqSaveAsTable`
-                  } else {
-                    val headParam = if (head.toString.contains("=")) {
-                      s"table = Table.Ref(${head.toString.split("=").last.trim})"
-                    } else {
-                      s"Table.Ref($head)"
-                    }
-                    val allArgs = (headParam :: tail).mkString(", ")
-                    Patch.replaceTree(a, s"$qual.saveAsBigQueryTable($allArgs)")
-                  }
+                  val paramsUpdated =
+                    params
+                      .zipWithIndex
+                      .foldLeft(List[Object]()) { case (accumulator, (param, index)) =>
```

Comment:
Why folding into a `List`. Can't we map ?

Reply:
What is the problem of folding in a list?